core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -393,10 +465,15 @@ public Builder withObjectMapper(ObjectMapper objectMapper) {
       return this;
     }
 
+    public Builder withRequestInterceptor(HttpRequestInterceptor interceptor) {
```

Comment:
Unless we need to make this public, I would rather not because it leaks the HttpRequestInterceptor through the Iceberg API. I think it can be private.